### ISSUE TITLE
Error: Failed to build the bibliography via biber

### ISSUE THREAD

**Issue by Dr-Joe-Roberts** [mitchelloharawild/vitae#103]
Hi,

Running the Marie Curie example of vitae (and my own CV) fails when producing a bibliography, resulting in the following:

```
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS Untitled.utf8.md --to latex --from markdown+autolink_bare_uris+tex_math_single_backslash --output Untitled.tex --self-contained --table-of-contents --toc-depth 2 --template "C:\Users\JOEROB~1\DOCUME~1\R\WIN-LI~1\3.6\vitae\RMARKD~1\TEMPLA~1\AWESOM~1\RESOUR~1\AWESOM~1.TEX" --number-sections --highlight-style tango --pdf-engine xelatex --biblatex --lua-filter "C:/Users/Joe Roberts/Documents/R/win-library/3.6/rmarkdown/rmd/lua/pagebreak.lua" --lua-filter "C:/Users/Joe Roberts/Documents/R/win-library/3.6/rmarkdown/rmd/lua/latex-div.lua" --wrap preserve --variable tables=yes --standalone --include-in-header "C:\Users\JOEROB~1\AppData\Local\Temp\RtmpKKOCtt\cv-header41ec38652259.tex" 
INFO - This is Biber 2.14
INFO - Logfile is 'Untitled.blg'
INFO - Reading 'Untitled.bcf'
INFO - Found 12 citekeys in bib section 0
INFO - Processing section 0
INFO - Globbing data source 'C:/Users/Joe Roberts/Documents/Untitled/curie.bib'
INFO - Globbed data source 'C:/Users/Joe Roberts/Documents/Untitled/curie.bib' to C:/Users/Joe\ Roberts/Documents/Untitled/curie.bib
INFO - Looking for bibtex format file 'C:/Users/Joe\ Roberts/Documents/Untitled/curie.bib' for section 0
ERROR - Cannot find 'C:/Users/Joe\ Roberts/Documents/Untitled/curie.bib'!
INFO - ERRORS: 1
Error: Failed to build the bibliography via biber
In addition: Warning message:
In RefManageR::as.BibEntry(.) :
  rownames of data.frame not meaningful for creating keys
Execution halted
Warning message:
LaTeX Warning: Empty bibliography on input line 216.
LaTeX Warning: Citation '1' undefined on input line 229.
LaTeX Warning: Citation '2' undefined on input line 229.
LaTeX Warning: Citation '3' undefined on input line 229.
LaTeX Warning: Citation '4' undefined on input line 229.
LaTeX Warning: Citation '5' undefined on input line 229.
LaTeX Warning: Citation '6' undefined on input line 229.
LaTeX Warning: Citation '7' undefined on input line 229.
LaTeX Warning: Citation '8' undefined on input line 229.
LaTeX Warning: Citation '9' undefined on input line 229.
LaTeX Warning: Citation '10' undefined on input line 229.
LaTeX Warning: Citation '11' undefined on input line 229.
LaTeX Warning: Citation '12' undefined on input line 229.
Package rerunfilecheck Warning: File `Untitled.out' has changed.
(rerunfilecheck)                Rerun to get outlines right
(rerunfilecheck)                or use package `bookmark'.
LaTeX Warning: There were undefined references.
LaTeX Warning: Label(s) may  [... truncated] 
```

It seems when globbing the data source it is adding an extra \  into the file path, but I'm not sure how or why. Any ideas?

**Comment by vlamenco** [comment]
same with me！

**Comment by tallguyjenks** [comment]
@Dr-Joe-Roberts  in your YAML header what is your bib file path? do you list out the whole C: drive and everything? if so, maybe try this file path `'~Documents/Untitled/curie.bib'` as the `~` defaults to *Documents* on windows.

**Comment by mitchelloharawild** [comment]
The extra `\` in the file path is to escape the space in your folder path.
I'm hopeful that this will be fixed with using multiple bib files via lua filters (#88)

**Comment by nspyrison** [comment]
I am running into the same thing.

@Dr-Joe-Roberts we aren't using a bib filepath in YAML. We call `bibliography_entries("rjhpubs.bib")` in an R chunk. 

Given a simple bib file: 
```
@software{spyrison_spinifex_2019,
	title = {spinifex: manual tours, manual control of dynamic projections of numeric multivariate data},
	url = {https://github.com/nspyrison/spinifex/},
	version = {0.1.0.9000},
	author = {Spyrison, Nicholas S. and Cook, Dianne},
	date = {2019}
}

@article{spyrison_spinifex_2020,
	title = {spinifex: An R Package for Creating a Manual Tour of Low-dimensional Projections of Multivariate Data},
	volume = {12},
	pages = {(accepted)},
	number = {1},
	journaltitle = {The R Journal},
	author = {Spyrison, Nicholas and {Dianne Cook}},
	date = {2020},
	file = {Spyrison - 2020 - spinifex An R Package for Creating a Manual Tour .pdf:C\:\\Users\\Nicholas Spyrison\\Zotero\\storage\\4PWPXRMD\\Spyrison - 2020 - spinifex An R Package for Creating a Manual Tour .pdf:application/pdf}
}
```

and in the .rmd I call 

```{r papers, eval=T}
bibliography_entries("nssPubs.bib") %>%
  filter(bibtype == "Article") %>%
  arrange(desc(date), title)
```
resulting in the .tex file having:
```
\defbibheading{bib-C:/Users/Nicholas Spyrison/Documents/R/CV/nssPubs.bib-4089067}{}
\addtocategory{bib-C:/Users/Nicholas Spyrison/Documents/R/CV/nssPubs.bib-4089067}{spyrison_spinifex_2020}
\newrefcontext[sorting=none]\setcounter{papers}{0}\pagebreak[3]
\printbibliography[category=bib-C:/Users/Nicholas Spyrison/Documents/R/CV/nssPubs.bib-4089067,heading=none]\setcounter{papers}{0}

\nocite{spyrison_spinifex_2020}
```
Which results in a similar error:
```
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS NicholasSpyrison_CV.utf8.md --to latex --from markdown+autolink_bare_uris+tex_math_single_backslash --output NicholasSpyrison_CV.tex --self-contained --table-of-contents --toc-depth 2 --template "C:\Users\NICHOL~1\DOCUME~1\R\WIN-LI~1\3.6\vitae\RMARKD~1\TEMPLA~1\hyndman\RESOUR~1\HYNDMA~1.TEX" --number-sections --highlight-style tango --pdf-engine pdflatex --biblatex --lua-filter "C:/Users/Nicholas Spyrison/Documents/R/win-library/3.6/rmarkdown/rmd/lua/pagebreak.lua" --lua-filter "C:/Users/Nicholas Spyrison/Documents/R/win-library/3.6/rmarkdown/rmd/lua/latex-div.lua" --wrap preserve --variable tables=yes --standalone -Mhas-frontmatter=false --include-in-header "C:\Users\NICHOL~1\AppData\Local\Temp\Rtmp6TB5zS\cv-header229450d14353.tex" 
INFO - This is Biber 2.14
INFO - Logfile is 'NicholasSpyrison_CV.blg'
INFO - Reading 'NicholasSpyrison_CV.bcf'
INFO - Found 1 citekeys in bib section 0
INFO - Processing section 0
INFO - Globbing data source 'C:/Users/Nicholas Spyrison/Documents/R/CV/nssPubs.bib'
INFO - Globbed data source 'C:/Users/Nicholas Spyrison/Documents/R/CV/nssPubs.bib' to C:/Users/Nicholas\ Spyrison/Documents/R/CV/nssPubs.bib
INFO - Looking for bibtex format file 'C:/Users/Nicholas\ Spyrison/Documents/R/CV/nssPubs.bib' for section 0
ERROR - Cannot find 'C:/Users/Nicholas\ Spyrison/Documents/R/CV/nssPubs.bib'!
INFO - ERRORS: 1
Error: Failed to build the bibliography via biber
Execution halted
Warning message:
LaTeX Warning: Empty bibliography on input line 382.
LaTeX Warning: Citation 'spyrison_spinifex_2020' undefined on input line 384.
Package rerunfilecheck Warning: File `NicholasSpyrison_CV.out' has changed.
(rerunfilecheck)                Rerun to get outlines right
(rerunfilecheck)                or use package `bookmark'.
LaTeX Warning: There were undefined references.
LaTeX Warning: Label(s) may have changed. Rerun to get cross-references right.
Package biblatex Warning: Please (re)run Biber on the file:
(biblatex)                NicholasSpyrison_CV
(biblatex)                and rerun LaTeX afterwards. 
```

**Comment by mitchelloharawild** [comment]
I've re-implemented the bibliography functionality to use pandoc's citeproc. Could you try this again for your issue?

**Comment by acircleda** [comment]
I'm having the same issue. I just installed Vitae and knit one of the default templates (with Marie Curie text) with no problem. Upon working with my own data, I have been running across this error where my PDF will knit but no bibliographic data is present:

```
Writing 2 Bibtex entries ... OK
Results written to file 'conf.bib'
output file: Untitled.knit.md

assertion failed: open my_perl
assertion failed: MZ magic bytes
assertion failed: PE header
assertion failed: IMAGE_NT_OPTIONAL_HDR_MAGIC
Error: Failed to build the bibliography via biber
In addition: Warning messages:
1: In RefManageR::as.BibEntry(.) :
  rownames of data.frame not meaningful for creating keys
2: In RefManageR::as.BibEntry(.) :
  rownames of data.frame not meaningful for creating keys
3: In RefManageR::as.BibEntry(.) :
  rownames of data.frame not meaningful for creating keys
4: In RefManageR::as.BibEntry(.) :
  rownames of data.frame not meaningful for creating keys
Execution halted
Warning message:
LaTeX Warning: Empty bibliography on input line 266.
LaTeX Warning: Citation '1' undefined on input line 268.
LaTeX Warning: Empty bibliography on input line 276.
LaTeX Warning: Citation '2' undefined on input line 278.
LaTeX Warning: Empty bibliography on input line 292.
LaTeX Warning: Citation '1' undefined on input line 300.
LaTeX Warning: Citation '2' undefined on input line 300.
LaTeX Warning: Citation '3' undefined on input line 300.
LaTeX Warning: Citation '4' undefined on input line 300.
LaTeX Warning: Citation '5' undefined on input line 300.
LaTeX Warning: Citation '6' undefined on input line 300.
LaTeX Warning: Citation '7' undefined on input line 300.
LaTeX Warning: Empty bibliography on input line 312.
LaTeX Warning: Citation '1' undefined on input line 315.
LaTeX Warning: Citation '2' undefined on input line 315.
Package rerunfilecheck Warning: File `Untitled.out' has changed.
(rerunfilecheck)                Rerun to get outlines right
(rerunfilecheck)                or u [... truncated] 
```

I am building my .bib files like so:

```{r}
library(gsheet)
options(encoding = "UTF-8")
gsheet2tbl("https://docs.google.com/spreadsheets/d/1qbMpXHXm41q69prUVjpQFfP_mMXjORNxFP_IbAoCuKo/edit#gid=318960497") %>%
  filter(category=="intl") %>%
  select(-category) %>%
  RefManageR::as.BibEntry() %>% 
  RefManageR::WriteBib("conf.bib")

bibliography_entries("conf.bib") 
```

Now, when I try to run the default Curie data, I get the same error. I've looked through previous issues and tried to follow the advice, but have not found a fix.

**Comment by mitchelloharawild** [comment]
Could you try with the dev version of the package? It uses a different method for generating the bibliography entries.

**Comment by acircleda** [comment]
Thanks for the fast response. I uninstalled the CRAN version, updated all package dependencies, and installed the Github version of Vitae. 

I now get this error:

```
Could not find bibliography file: C:
pandoc.exe: PandocFilterError "pandoc-citeproc" "Filter returned error status 1"
Error: pandoc document conversion failed with error 1
```
If I remove the bib, it knits fine.

**Comment by mitchelloharawild** [comment]
Could you post the generated `md` file when `keep_md: true` is set?
```
output: 
  vitae::your_vitae_style:
    keep_md: true
```

**Comment by acircleda** [comment]
It took a lot of tries, as the file kept getting deleted despite `true` being indicated.

```
---
name: Marie
surname: Curie
position: "Professor"
address: "School of Physics & Chemistry, École Normale Supérieure"
phone: +1 22 3333 4444
www: mariecurie.com
email: "Marie.Curie@ens.fr"
twitter: mariecurie
github: mariecurie
linkedin: mariecurie
date: "May 2020"
output: 
  vitae::hyndman:
    keep_md: true
---



# Some stuff about me

 * I poisoned myself doing research.
 * I was the first woman to win a Nobel prize
 * I was the first person and only woman to win a Nobel prize in two different sciences.

# Education

\detailedsection{\detaileditem{Informal studies}{1889-91}{Flying University}{Warsaw, Poland}{\empty}\detaileditem{Master of Physics}{1893}{Sorbonne Université}{Paris, France}{\empty}\detaileditem{Master of Mathematics}{1894}{Sorbonne Université}{Paris, France}{\empty}}

# Nobel Prizes

\briefsection{\briefitem{Nobel Prize in Physics}{1903}{Awarded for her work on radioactivity with Pierre Curie and Henri Becquerel}\briefitem{Nobel Prize in Chemistry}{1911}{Awarded for the discovery of radium and polonium}}

# Publications


::: {#bibliography}
C:\Users\aschmidt\AppData\Local\Temp\RtmpyqjPaj\file26b4589857d1.yaml
:::

```

**Comment by mitchelloharawild** [comment]
Thanks for your persistence - it looks like the file is being created at a reasonable location (
`C:\Users\aschmidt\AppData\Local\Temp\RtmpyqjPaj\file26b4589857d1.yaml`), however I wonder if pandoc-citeproc expects `\` to be `/`. I've just pushed this change in 0454bc857062c4b6b70365c339b66cbaf38be6a2, could you update and try again?

**Comment by acircleda** [comment]
Same issue. Here is my session info, FWIW. Realize I need to upgrade my `R` version but nervous to do so:

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
 [1] RefManageR_1.2.12  forcats_0.4.0      stringr_1.4.0     
 [4] dplyr_0.8.5        purrr_0.3.3        readr_1.3.1       
 [7] tidyr_1.0.0        tibble_3.0.1       ggplot2_3.3.0.9000
[10] tidyverse_1.3.0    gsheet_0.4.2      

loaded via a namespace (and not attached):
 [1] nlme_3.1-140       matrixStats_0.55.0 fs_1.3.1          
 [4] usethis_1.5.1      lubridate_1.7.8    scholar_0.1.7     
 [7] devtools_2.1.0     httr_1.4.1         rprojroot_1.3-2   
[10] rstan_2.19.2       R.cache_0.14.0     tools_3.6.1       
[13] backports_1.1.7    utf8_1.1.4         R6_2.4.0          
[16] DBI_1.1.0          colorspace_1.4-1   withr_2.1.2       
[19] tidyselect_1.1.0   gridExtra_2.3      prettyunits_1.0.2 
[22] processx_3.4.1     curl_4.3           compiler_3.6.1    
[25] cli_2.0.2          ezpickr_2.0.0      rvest_0.3.5       
[28] vitae_0.2.2.9000   xml2_1.3.2         desc_1.2.0        
[31] scales_1.1.0       callr_3.3.1        digest_0.6.25     
[34] StanHeaders_2.19.0 rmarkdown_2.1      R.utils_2.9.2     
[37] pkgconfig_2.0.3    htmltools_0.4.0    sessioninfo_1.1.1 
[40] bibtex_0.4.2.2     dbplyr_1.4.2       rlang_0.4.6       
[43] readxl_1.3.1       rstudioapi_0.11    generics_0.0.2    
[46] jsonlite_1.6.1     R.oo_1.23.0        inline_0.3.15     
[49] magrittr_1.5       loo_2.1.0          Rcpp_1.0.4.6      
[52] munsell_0.5.0      fansi_0.4.0        lifecycle_0.2.0   
[55] R.methodsS3_1.7.1  stringi_1.4.6      yaml_2.2.1        
[58] pkgbuild_1.0.4     plyr_1.8.6         grid_3.6.1        
[61] parallel_3.6.1     crayon_1.3.4       lattice_0.20-38   
[64] haven_2.2.0        hms_0.5.3          knitr_1.28        
[67] ps_1.3.0           pillar_1.4.4       stats4_3.6.1      
[70] pkgload_1.0.2      reprex_0.3.0       glue_1.4.1        
[73] packrat_0.5.0      evaluate_0.14      remotes_2.1.0     
[76] modelr_0.1.5       vctrs_0.3.0        testthat_2.2.1    
[79] cellranger_1.1.0   gtable_0.3.0       assertthat_0.2.1  
[82] xfun_0.14          broom_0.5.6        memoise_1.1.0     
[85] ellipsis_0.3.1  
```

**Comment by mitchelloharawild** [comment]
Maybe the file is being deleted before citeproc can access it on Windows? Or possibly the complete file paths are causing trouble. I've made another guess in 128246082b4ecdf2b1820f599e3f3d9d214d9ac2, which should result in a very similar lua filter call to what worked for you in https://github.com/jooyoungseo/jy_CV

Could you update and try again?

**Comment by acircleda** [comment]
Success! It knitted with no problems. Thank you very much!

**Comment by mitchelloharawild** [comment]
Great to hear!
Unfortunately it means that the intermediate bib file will be stored in the CV's working directory, but it'll work for now.

**Comment by Nasser441** [comment]
install.packages('vitae')

**Comment by Nasser441** [comment]
[]()@mitchelloharawild

**Comment by mitchelloharawild** [comment]
> @mitchelloharawild

?

**Comment by mitchelloharawild** [comment]
> Thanks for the fast response. I uninstalled the CRAN version, updated all package dependencies, and installed the Github version of Vitae.
> 
> I now get this error:
> 
> ```
> Could not find bibliography file: C:
> pandoc.exe: PandocFilterError "pandoc-citeproc" "Filter returned error status 1"
> Error: pandoc document conversion failed with error 1
> ```
> 
> If I remove the bib, it knits fine.
> 
> Just for a little more context, I also forked https://github.com/jooyoungseo/jy_CV, which uses `vitae` but calls .bib through YAML not `bibliography_entries`. It can knit the .bib files fine. However, if I try to use `bibliography_entries`, it gets the same error above.

This has now been fixed in fc7398b60f557c97f0144a465bbadd28c3242cb8

Closing as bibliography is now generated via pandoc and pandoc-citeproc, so the behaviour of bibliography should closely match what is done with typical rmarkdown documents.
